mobile/examples/object_detections/iOS/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import AVFoundation
+import Accelerate
+import CoreImage
+import Darwin
+import Foundation
+import UIKit
+
+
+// Result struct
+struct Result {
+    let processTimeMs: Double
+    let inferences: [Inference]
+}
+
+// Inference struct for ssd model
+struct Inference {
+    let score: Float
+    let className: String
+    let rect: CGRect
+    let displayColor: UIColor
+}
+
+// Information about a model file or labels file.
+typealias FileInfo = (name: String, extension: String)
+
+enum OrtModelError : Error {
+    case error(_ message: String)
+}
+
+class ModelHandler: NSObject {
+    
+    // MARK: - Inference Properties
+    let threadCount: Int32
+    let threshold: Float = 0.5
+    let threadCountLimit = 10
+    
+    // MARK: - Model Parameters
+    let batchSize = 1
+    let inputChannels = 3
+    let inputWidth = 300
+    let inputHeight = 300
+    
+    private let colors = [
+        UIColor.red,
+        UIColor(displayP3Red: 90.0/255.0, green: 200.0/255.0, blue: 250.0/255.0, alpha: 1.0),
+        UIColor.green,
+        UIColor.orange,
+        UIColor.blue,
+        UIColor.purple,
+        UIColor.magenta,
+        UIColor.yellow,
+        UIColor.cyan,
+        UIColor.brown
+    ]
+    
+    private var labels: [String] = []
+    
+    init?(threadCount: Int32 = 1) {
+
+        self.threadCount = threadCount
+        
+        super.init()
+    }
+
+    // This methods preprocess the image, runs the ort inferencesession and returns the inference result
+    func runModel(onFrame pixelBuffer: CVPixelBuffer, modelFileInfo: FileInfo, labelsFileInfo: FileInfo) throws -> Result? {
```

Comment:
general comment: I noticed there are a few different error reporting modes - e.g., returning nil/printing, throwing, calling fatalError(). I'm not too familiar with Swift conventions. is there a reason for using each of the different modes?

Reply:
I am not sure there really exists an error handling convention in Swift or not. But in general I think 
- fatalError() for my case is only used for failures like AVCameraSession failure/videodeviceinputoutput/resourcefile fetch failure, which can cause a crash for the app. Part of this is directly kept from apple AVCamera Sample app.
- for printing/return nil, it is used in pair with `guard let`, which is for some variable type match, value checking, verify conditions, etc. Or with do/catch handling.
-  for throw/throws, I believe it is mainly used just for throwing OrtModelError in the app.

Reply:
I guess what I'm asking is - could it be more consistent or is there a good reason to do all three?

Reply:
oh okay, sorry about the misunderstanding. Maybe I can try to only use the previous two to make it more consistent and reduces the number of printing messages that can cause some confusion.